frontend/tests/features/wheresMyGene/landingPage.test.ts

```diff
@@ -0,0 +1,137 @@
+    await expect(page.getByTestId(ADD_TISSUE_LBL)).toContainText("Add Tissues");
+
+    // STEP 2 column
+    await expect(page.getByTestId("column-two")).toBeVisible();
+    await expect(page.getByTestId(ADD_GENE_LBL)).toContainText("STEP 2");
+    await expect(page.getByTestId(ADD_GENE_LBL)).toContainText("Add Genes");
+
+    // STEP 3 column
+    await expect(page.getByTestId(EXPLORE_GENE_EXPRESSION)).toContainText(
+      "STEP 3"
+    );
+    await expect(page.getByTestId(EXPLORE_GENE_EXPRESSION)).toContainText(
+      "Explore Gene Expression"
+    );
+
+    // Download
+    await expect(page.getByTestId(LEGEND_WRAPPER)).toContainText("Download");
+    await expect(page.getByTestId("download-button")).toBeDisabled();
+
+    // Share
+    await expect(page.getByTestId(LEGEND_WRAPPER)).toContainText("Share");
+    await expect(page.getByTestId("share-button")).toBeDisabled();
+
+    // Source data
+    await expect(page.getByTestId(LEGEND_WRAPPER)).toContainText("Source Data");
+    await expect(
+      page.getByTestId("source-data-button").locator("svg")
+    ).toBeEnabled();
+
+    // Gene expression
+    await expect(page.getByTestId(LEGEND_WRAPPER)).toContainText(
+      "Gene Expression"
+    );
+    await expect(
+      page.locator('[id="visualization-color-scale"]')
+    ).toBeVisible();
+
+    // Gene expression in cells
+    await expect(page.getByTestId(LEGEND_WRAPPER)).toContainText(
+      "Expressed in Cells (%)"
+    );
+    await expect(page.locator('[id="expressed-in-cells-dots"]')).toBeVisible();
+    const actualSizes = [];
+    for (let i = 0; i < 5; i++) {
+      actualSizes.push(
+        Number(
+          await page
+            .locator('[id="expressed-in-cells-dots"]')
+            .locator("span")
+            .nth(i)
+            .getAttribute("size")
+        )
+      );
+    }
+    expect(actualSizes).toEqual(DOT_SIZES);
+  });
+  test("Should verify top nav", async ({ page }) => {
+    const DOCUMENTATION = "Documentation";
+    await goToWMG(page);
+    // verify logo
+    expect(page.getByTestId("logo")).toBeVisible();
+
+    // Help & Doc
+    expect(await page.getByTestId("InputDropdown").textContent()).toBe(
+      "Help & Documentation"
+    );
+
+    await page.getByTestId("InputDropdown").click();
+
+    const popupPromise = page.waitForEvent("popup");
+    await page.getByText(DOCUMENTATION, { exact: true }).click();
+    const popup = await popupPromise;
+    // Wait for new tab to load.
+    await popup.waitForLoadState();
+    expect(popup.url()).toContain(DOCUMENTATION);
+    expect(
+      popup.locator(
+        getById("gene-expression--query-gene-expression-across-tissues")
+      )
+    ).toBeVisible();
+  });
+  test("Should logo takes user to portal page", async ({ page }) => {
+    await goToWMG(page);
+
+    await page.getByTestId("logo").click();
+    await page.waitForLoadState();
+    expect(page.getByTestId("laptop-with-cell-data-on-screen")).toBeVisible();
```

Comment:
not sure if it helps to add tryUntil wrapper?

I have a PR here PTAL thank you!

https://github.com/chanzuckerberg/single-cell-data-portal/pull/4549/files